contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
+          unit: NX_ANY
+        extent(NX_FLOAT):
+          exists: recommended
+        associated_source(NX_CHAR):
+          exists: recommended
+          doc: |
+            The path to a the source, which emitted this beam.
+            Should be named with the same appendix as the source, e.g.,
+            for TYPE=532nmlaser, there should as well be
+            a NXsource named "source_532nmlaser" aside of this Beam
+            instance named "beam_532nmlaser"
+            
+            Example: /entry/instrument/source_532nmlaser
+        # The two polarization descriptions may be completely replaced by
+        # incident_polarization
+        beam_polarization_type:
+          exists: optional
+          enumeration: [linear, circular, ellipically, unpolarized]
+        linear_beam_sample_polarization(NX_NUMBER):
+          exists: optional
+          doc: |
+            Description of the light polarization of the respective beam and the beam-sample-normal.
+            This is only well defined, if the beam direclty hits the sample (either indicent
+            beam or detection beam) and if the samples needs to have a clearly
+            defined sample surface. The beam-sample-normal plane is spanned by
+            the beam and the samples surface normal.
+
+            If the electric field in in this plane, this value is 0° (P-pol)
+            If the electric field is perpendicular to this plane, this value is 90° (S-pol)
+            If the surface normal is parallel to the beam direction, then only the
+            relation between different polarizations has to be consistent 
+            (i.e. parallel polarized 0° and cross polarized 90°)
+          unit: NX_ANGLE
+      # Discussion with florian:
+      # The user wants to use simple parameters, which they are used to.
+      # Though, via NXtransformations in principle maximum flexibility can be provided,
+      # even for unusual and complex experimental setups.
+      # The solution from mpes was: simple "front end" by only stating one angle
+      # with the "backend" to be customizable via NXtransformations.
+      #
+      # fundamental orientation of incident and detection beam
       angle_of_incidence(NX_NUMBER):
+      # relation beam directions (incident and detection) with sample orientation
+      # these angles are quite simple. They might be extended similar to
+      # NXmpes_liquid with leaf_normal to allow arbitrary description.
+      # this might require a different reference frame.
+        exists: optional
         unit: NX_ANGLE
         doc: |
           Angle(s) of the incident beam vs. the normal of the bottom reflective
-          (substrate) surface in the sample.
+          (substrate) surface in the sample. These two directions span the plane
+          of incidence.
         dimensions:
           rank: 1
           dim: [[1, N_incident_angles]]
         \@units:
-      detection_angle(NX_NUMBER):
+      angle_of_detection(NX_NUMBER):
```

Comment:
Consider using `NXtransformations` for describing theses angles.

Reply:
This is planned but will be implemented later, if the reference frame discussion is done